redis/src/main/scala/zio/redis/api/PubSub.scala

```diff
@@ -0,0 +1,41 @@
+package zio.redis.api
+
+import zio.redis.Input._
+import zio.redis.Output._
+import zio.redis._
+import zio.{ Chunk, ZIO }
+
+trait PubSub {
+  import PubSub._
+
+  final def pSubscribe(a: String, as: String*): ZIO[RedisExecutor, RedisError, Unit] = PSubscribe.run((a, as.toList))
+
+  final def pubSubChannels(a: String): ZIO[RedisExecutor, RedisError, Chunk[String]] =
+    PubSubChannels.run(("CHANNELS", List(a)))
+
+  final def pubSubNumSub(a: String, as: String*): ZIO[RedisExecutor, RedisError, Map[String, Long]] =
+    PubSubNumSub.run(("NUMSUB", a :: as.toList))
+
+  final def pubSubNumPat(): ZIO[RedisExecutor, RedisError, Long] = PubSubNumPat.run("NUMPAT")
+
+  final def publish(a: String, b: String): ZIO[RedisExecutor, RedisError, Long] = Publish.run((a, b))
+
+  final def pUnSubscribe(as: String*): ZIO[RedisExecutor, RedisError, Unit] = PUnSubscribe.run(as.toList)
+
+  final def subscribe(a: String, as: String*): ZIO[RedisExecutor, RedisError, Unit] = Subscribe.run((a, as.toList))
```

Comment:
The problem with `subscribe` is that it keeps an open connection while receiving messages. This could be represented with `ZStream` instead of `ZIO`. I recommend that you try using it through telnet, and then evaluate what changes will be required in order to implement this. You can write your conclusion in issue #160.

Reply:
kk